### PR TITLE
Add Diagnostics Events for Device, Maintenance, and Settings

### PR DESCRIPTION
**Pull request by sakimsft**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Add diagnostic events for the following categories:
-Device
-Maintenance
-Settings

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-remote-monitoring-webui/1196)

### CODE REVIEW

File: src/components/shell/flyouts/settings/settings.js [hunk 1]
```diff
@@ -157,7 +175,7 @@ export class Settings extends LinkedComponent {
         <Flyout.Container>
           <Flyout.Header>
             <Flyout.Title>{t('settingsFlyout.title')}</Flyout.Title>
-            <Flyout.CloseBtn onClick={onClose} />
+            <Flyout.CloseBtn onClick={this.onFlyoutClose.bind(this, 'Settings_TopXClose_Click')} />
```
Comment (molly-moen) [inline src/components/shell/flyouts/settings/settings.js]: I don't think you need to call bind when you use the = (...) => syntax

File: src/components/shell/flyouts/settings/settings.js [hunk 2]
```diff
@@ -69,21 +70,39 @@ export class Settings extends LinkedComponent {
...
       [name]: value
+    },
+    () => {
+      this.props.logEvent(toSinglePropertyDiagnosticsModel('Settings_SimulationToggle', 'isEnabled', toggledSimulation));
     });
     this.props.toggleSimulationStatus(etag, value);
   }
 
+  onThemeChange = (nextTheme) => {
+    this.props.logEvent(toDiagnosticsModel('Settings_ThemeChanged', {}));
```
[the remark this answers is not in the document]
Comment (sakimsft) [reply src/components/shell/flyouts/settings/settings.js]: according to requirement, no metadata required.

File: src/components/pages/devices/devices.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ export class Devices extends Component {
   openSIMManagement = () => this.setState({ openFlyoutName: 'sim-management' });
   openNewDeviceFlyout = () => {
     this.setState({ openFlyoutName: 'new-device' });
-    this.props.logEvent(toDiagnosticsModel('Devices_NewClick', {}));
+    this.props.logEvent(toDiagnosticsModel('Devices_New_Click', {}));
```
[the remark this answers is not in the document]
Comment (sakimsft) [reply src/components/pages/devices/devices.js]: Are we going to keep using the existing app insights as final deliverable?  
I thought we are going to revisit presentation part at phase 2.

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.js [hunk 4]
```diff
@@ -269,7 +269,7 @@ export class DeviceNew extends LinkedComponent {
   }
 
   deviceChange = ({ target: { value } }) => {
-    this.props.logEvent(toSinglePropertyDiagnosticsModel('Devices_DeviceSelect', 'Device',
+    this.props.logEvent(toSinglePropertyDiagnosticsModel('Devices_Device_Select', 'Device',
```
[the remark this answers is not in the document]
Comment (sakimsft) [reply src/components/pages/devices/flyouts/deviceNew/deviceNew.js]: same comment.

File: src/components/pages/devices/devices.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sakimsft) [reply src/components/pages/devices/devices.js]: Who is using the existing app insights?

File: src/components/shell/flyouts/settings/settings.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sakimsft) [reply src/components/shell/flyouts/settings/settings.js]: Agreed and good comment. confirmed changing requirement to add metadata.

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.js [hunk 5]
```diff
@@ -292,11 +292,23 @@ export class DeviceNew extends LinkedComponent {
   }
 
   deviceTypeChange = ({ target: { value } }) => {
-    this.props.logEvent(toSinglePropertyDiagnosticsModel('Devices_DeviceTypeSelect', 'DeviceType',
+    this.props.logEvent(toSinglePropertyDiagnosticsModel('Devices_DeviceType_Select', 'DeviceType',
```
Comment (molly-moen) [inline src/components/pages/devices/flyouts/deviceNew/deviceNew.js]: couple more instances of name changes to be reverted: DeviceTypeSelect, ApplyClick, TopXCloseClick,, CancelClick, CloseClick

**Pull request merged**